homework1/src/app/game-state.service.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable } from "@angular/core";
+import { GameState } from "./game-state";
+
+@Injectable({
+  providedIn: "root",
+})
+export class GameStateService {
+  gameState: GameState = {
+    state: "start",
+    gameStart: false,
+    gameFinished: false,
+  };
+
+  constructor() {}
+
+  getStates(): GameState {
+    return this.gameState;
+  }
+  setStates(prop: string, value: string | boolean): void {
```

Comment:
it is better to divide in into two methods, one that will set game start and finished, and the other that will change the state.For the start/finish method just provide a boolean to know which field `gameStart` or `gameFinished` to update. in this way you can avoid using `string | boolean` and setting `this.gameState[prop] ` like this.